server/src/main/java/org/opensearch/index/shard/CloseableRetryableRefreshListener.java

```diff
@@ -30,7 +31,9 @@ public abstract class CloseableRetryableRefreshListener implements ReferenceMana
      * Total permits = 1 ensures that there is only single instance of performAfterRefresh that is running at a time.
      * In case there are use cases where concurrency is required, the total permit variable can be put inside the ctor.
      */
-    private static final int TOTAL_PERMITS = 1;
+    private static final int TOTAL_PERMITS = Integer.MAX_VALUE;
```

Comment:
Is the recommendation in the comment (`In case there are use cases where concurrency is required, the total permit variable can be put inside the ctor.`) not valid anymore? Should we remove it?

Reply:
Yes, this comment is not required anymore.